go/state/gostate/archive_state.go

```diff
@@ -155,14 +155,19 @@ func (s *ArchiveState) Export(ctx context.Context, out io.Writer) (common.Hash,
 		return common.Hash{}, state.ExportNotSupported
 	}
 
-	exportableTrie := mptio.NewExportableArchiveTrie(trie, s.block)
-	rootHash, err := mptio.ExportLive(ctx, mptio.NewLog(), exportableTrie, out)
+	err := mptio.ExportBlockFromOnlineArchive(ctx, nil, trie, out, s.block)
```

Comment:
You can return the hash from this func and remove it from getting on line 164.

Reply:
I am not fan of the parent method design at the first place.  I think API should provide the simplest (most straightforward) use-cases.  I want export - there is a method to export, I need a root hash - there is a method to get it (we already have it),

but having a method that "exports a database and as a side effect gives hash of the original database" is too complicated - it servers two purposes. 

As a consequence, there will be clients that must ignore one of the results because it does not serve their needs.  

I.e. I do not want to corrupt API of `ExportBlockFromOnlineArchive` to fit this particular callsite